onlineShop/Data/Database/DBStocksProvider.cs

```diff
@@ -0,0 +1,77 @@
+﻿using onlineShop.Data.Entities;
+using System;
+using System.Linq;
+
+namespace onlineShop.Data.Database
+{
+    public class DBStocksProvider : IStocksProvider
+    {
+        private StocksModel _stocksModel;
+        public DBStocksProvider()
+        {
+            _stocksModel = new StocksModel();
+        }
+
+        public int GetProductStock(Guid productId)
+        {
+            var stock = _stocksModel.Stocks.FirstOrDefault(s => s.ProductId == productId);
+            if (stock == null)
+            {
+                return 0;
+            }
+
+            return stock.Amount;
+        }
+
+        public bool TryAddStock(Guid productId, int count)
+        {
+            var stock = new Stock();
+            stock.Amount = count;
+            stock.ProductId = productId;
+            _stocksModel.Stocks.Add(stock);
+            _stocksModel.SaveChanges();
+
+            return true;
+        }
+
+        public bool TryDecreaseStock(Guid productId, int count)
+        {
+            var stock = _stocksModel.Stocks.FirstOrDefault(s => s.ProductId == productId);
+            if (stock == null)
+            {
+                return false;
+            }
+
+            stock.Amount -= count;
+            _stocksModel.SaveChanges();
+
+            return true;
+        }
+
+        public bool TryIncreaseStock(Guid productId, int count)
+        {
+            var stock = _stocksModel.Stocks.FirstOrDefault(s => s.ProductId == productId);
+            if (stock == null)
+            {
+                return false;
+            }
+
+            stock.Amount += count;
```

Comment:
This does not work if Amount is unlimited(-1). Instead of adding the count, it should keep the Amount as unlimited.